src/msg/async/ProtocolV1.cc

```diff
@@ -341,19 +341,27 @@ void ProtocolV1::write_event() {
         prepare_send_message(connection->get_features(), m, data);
       }
 
-      r = write_message(m, data, more);
+      write_message(m, data, more);
 
       connection->write_lock.lock();
-      if (r == 0) {
-        ;
-      } else if (r < 0) {
-        ldout(cct, 1) << __func__ << " send msg failed" << dendl;
-        break;
-      } else if (r > 0)
-        break;
     } while (can_write == WriteStatus::CANWRITE);
     connection->write_lock.unlock();
 
+    //batch handle sendmsg
+    {
+      ssize_t total_send_size = connection->outcoming_bl.length();
+      r = connection->_try_send(more);
+      if (r < 0) {
```

Comment:
Is there something similiar to Linux likely/unlikely marcos that can be used here? It's unlikely that 'r < 0', right?